### PR TITLE
Omnibus

### PR DESCRIPTION
**Pull request by perrydv**
This PR fixes six old and/or minor issues:

1. Issue #378: I made options(error = recover) work for the main case of interest: size processing called from `RCfunProcessing::process`.  Proposed solution is to defer to R’s native handling if options(‘error’) is non-null, and otherwise to use our existing  `tryCatch` clause.

2. Issue #374: I removed the `MCMC_*` files from the repo above the R package level.  According to the issue, @paciorek has these files saved separately for when they are needed but they do not need to live in the repo.

3. Issue #365: support for keyword `next` was added. A test was added to `test-misc.R`.

4. Issue #314: `NaN` is now successfully written to C++ from R via a `nimble_NaN()` function that attempts to use `std::numeric_limits::quiet_NaN()` when available and otherwise computes a NaN as 0./0.  A test was added to `test-misc.R`.

5. Issue #302: dots in R variable names in the DSL (e.g. `a.b`) are now handled by processing through our heavily used utility `Rname2CppName`. This occurs at the final step of writing C++ code.  The mangled result is `a_dot_b`.

6. Issue #92: This old issue was to support `pi` in the DSL.  Brief discussion led to a sticking point and it was left to go stale.  In R, `pi` exists as a constant that can be overwritten by the user.  Proposed solution to yield similar compiled behavior is that if a variable `pi` is used, it is always initialized to the natural constant `pi`, after which it behaves like any other local variable and hence can be overwritten.  Behavior will be limited by static typing: if used first on RHS, it will be a scalar double and can only be over-written as such.  Three tests were added to `test-misc.R`.

**Note that `pi` still does not work in model code.** We can open that as a separate issue if desired: it involves model processing and was not handled as part of Issue #92.

### CODE REVIEW

File: packages/nimble/inst/CppCode/Utils.cpp [hunk 1]
```diff
@@ -67,3 +67,9 @@ double pairmin(double x1, double x2) {return(x1 < x2 ? x1 : x2);}
 int nimStep(double x) { return(x >= 0 ? 1 : 0);}
 double cube(double x) {return(x*x*x);}
 double inprod(double v1, double v2) {return(v1*v2);}
+
+double nimble_NaN() {
```
Comment (fritzo) [inline packages/nimble/inst/CppCode/Utils.cpp]: Consider making this inline and optimizing
```c++
inline double nimble_NaN() {
  return std::numeric_limits<double>::has_quiet_NaN
    ? std::numeric_limits<double>::quiet_NaN()
    : (0. / 0.);
}
```

File: packages/nimble/R/RCfunction_compile.R [hunk 2]
```diff
@@ -237,26 +237,29 @@ RCfunProcessing <- setRefClass('RCfunProcessing',
...
-                                                                         sep = '\n')
-                                                    }
-                                                    stop(message, call. = FALSE)
-                                                })
+
+                                       ## exprClasses_setSizes contains lots of reticulated error trapping.
+                                       ## If options('error') is not NULL (typically options(error = recover)), let that take precedent for error trapping of exprClasses_setSizes.
+                                       ## Otherwise, wrap the setSizes call in our own error-trapping that outputs: any local message from a stop() inside a size handler,
+                                       ##    a message from here showing the larger block of code in which the error occurred, and, if nimbleOptions('verboseErrors') is TRUE, the call stack. 
+                                       if(!is.null(options('error'))) exprClasses_setSizes(compileInfo$nimExpr, compileInfo$newLocalSymTab, compileInfo$typeEnv)
```
Comment (fritzo) [inline packages/nimble/R/RCfunction_compile.R]: Good idea! Maybe we can get rid of `verboseErrors` now.

**Comment by fritzo** [comment]
:+1:

**Comment by fritzo** [comment]
Remember to move `nimble_NaN()` from to the header file if you make it inline.

**Comment by paciorek** [comment]
No concerns on my part; didn't look very deeply.

**Pull request merged**